modules/dnn/src/model.cpp

```diff
@@ -1499,5 +1499,770 @@ int TextDetectionModel_DB::getMaxCandidates() const
+{
+    std::static_pointer_cast<FaceDetectionModel_SSD_Impl>(impl)->setInputCrop(crop);
+    return *this;
+}
+
+FaceDetectionModel_SSD& FaceDetectionModel_SSD::setInputSwapRB(bool swapRB)
+{
+    std::static_pointer_cast<FaceDetectionModel_SSD_Impl>(impl)->setInputSwapRB(swapRB);
+    return *this;
+}
+
+void FaceDetectionModel_SSD::setInputParams(double scale, const Size& size, const Scalar& mean, bool swapRB, bool crop)
+{
+    std::static_pointer_cast<FaceDetectionModel_SSD_Impl>(impl)->setInputParams(scale, size, mean, swapRB, crop);
+}
+
+struct FaceDetectionModel_YN_Impl : public FaceDetectionModel_Impl
+{
+private:
+    float confThreshold;
+    float nmsThreshold;
+    Size inputSize;
+    std::vector<Rect2f> priors;
+    std::vector<std::vector<Point>> landmarks;
+
+public:
+    FaceDetectionModel_YN_Impl()
+        : confThreshold(0.9f),
+          nmsThreshold(0.3f),
+          inputSize(Size(-1, -1))
+    {
+        CV_TRACE_FUNCTION();
+    }
+
+    FaceDetectionModel_YN_Impl(const Net& network)
+        : FaceDetectionModel_YN_Impl()
+    {
+        CV_TRACE_FUNCTION();
+        initNet(network);
+    }
+
+    virtual
+    void detect(
+        InputArray frame,
+        CV_OUT std::vector<float>& confidences,
+        CV_OUT std::vector<Rect>& boxes,
+        float confThreshold = 0.9f,
+        float nmsThreshold = 0.3f
+    ) CV_OVERRIDE
+    {
+        CV_TRACE_FUNCTION();
+
+        // Re-Generate priors when input different size image.
+        if (inputSize != frame.getMat().size())
+        {
+            inputSize = frame.getMat().size();
+            setInputSize(inputSize);
+            generatePriors();
+        }
+
+        // Run inference process and post process.
+        std::vector<Mat> output_blobs;
+        processFrame(frame, output_blobs);
+
+        Mat outputs = postProcess(output_blobs);
+        outputs.convertTo(outputs, CV_32FC1);
+
+        // Format for outputs.
+        boxes.clear();
+        confidences.clear();
+        landmarks.clear();
+
+        for (int i = 0; i < outputs.rows; i++)
+        {
+            Mat output = outputs.row(i);
+
+            const int x = static_cast<int>(output.at<float>(0));
+            const int y = static_cast<int>(output.at<float>(1));
+            const int width = static_cast<int>(output.at<float>(2));
+            const int height = static_cast<int>(output.at<float>(3));
+            boxes.push_back(Rect(x, y, width, height));
+
+            std::vector<Point> landmark;
+            for (int j = 4; j < 14; j+=2)
+            {
+                const int x = static_cast<int>(output.at<float>(j));
+                const int y = static_cast<int>(output.at<float>(j + 1));
+                landmark.push_back(Point(x, y));
+            }
+            landmarks.push_back(landmark);
+
+            const float conficence = output.at<float>(14);
+            confidences.push_back(conficence);
+        }
```

Comment:
This block of formatting outputs can be blended into `postProcess` to reduce redundancy.

Reply:
I ported it as-is from your original implement. However as you say, it is more efficient to incorporate it in post process function. I'll consider and implement it. Thanks,

Reply:
I've implemented it! 👍